File: Java/Lab1/Lab1_task2.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        // Более компактная инициализация
        List<String> elements = new ArrayList<>(Arrays.asList("a", "1", "b", "2", "c", "3"));
        List<String> letters = new ArrayList<>();
        List<String> numbers = new ArrayList<>();

        // Копируем все элементы в оба списка
        letters.addAll(elements);
        numbers.addAll(elements);
        
        elements.clear(); // очищаем исходный список
        
        // Выводим буквы (элементы с четными индексами)
        for (int i = 0; i < letters.size(); i += 2) {
            System.out.print(letters.get(i));
        }
        System.out.println();
        
        // Выводим числа (элементы с нечетными индексами)
        for (int i = 1; i < numbers.size(); i += 2) {
            System.out.print(numbers.get(i));
        }
    }
}

File: C++/Lab1/Lab1_task2.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
	vector<string> elements = { "a", "1", "b", "2", "c", "3" };				   // исходный массив
	vector<string> letters;													   // массив букв
	vector<string> numbers;													   // массив чисел

	for (int i = 0; i < elements.size(); ++i) letters.push_back(elements[i]);  // заполнили массив буквами
	for (int i = 0; i < elements.size(); ++i) numbers.push_back(elements[i]);  // заполнили массив числами
	elements.clear();														   // очищаем исходный массив
	for (int i = 0; i < letters.size(); i+=2) cout << letters[i];					   // выводим первый массив
	cout << "\n";															   
	for (int i = 1; i < numbers.size(); i+=2) cout << numbers[i];					   // выводим второй массив

	return 0;
}
